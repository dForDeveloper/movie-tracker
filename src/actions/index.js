export const addMovie = (movie) => ({
  type: 'ADD_MOVIE',
  movie
});

export const setUser = (user) => ({
  type: 'SET_USER',
  user
});

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
});

export const getFavorites = (favorites) => ({
  type: 'GET_FAVORITES',
  favorites
});

export const addFavorite = (movie) => ({
  type: 'ADD_FAVORITE',
  movie
});