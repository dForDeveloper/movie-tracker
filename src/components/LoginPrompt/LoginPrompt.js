import React from 'react';
import { Link } from 'react-router-dom';

const LoginPrompt = () => {
  return (
    <div className="login-prompt">
      <Link to='/'><button>X</button></Link>
      <h3 className="prompt-text">
        You must be logged in to save favorites.
      </h3>
      <Link to='/sign-up'><button>Sign Up</button></Link>
      <Link to='/login'><button>Log In</button></Link>
    </div>
  );
}

export default LoginPrompt;