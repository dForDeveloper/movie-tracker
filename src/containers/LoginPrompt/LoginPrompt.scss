.LoginPrompt {
  z-index: 1;
  top: 30%;
  left: 35%;
  height: 170px;
  width: 430px;
  position: absolute;
  display: inline-block;
  overflow: hidden;
  text-align: center;
  background-color: rgba(243, 241, 255, 0.9);
  border: 6px solid $steel-grey;
}

.close-button,
.loginprompt-button,
.signupprompt-button {
  font-weight: bold;
  border: 2px solid $border-color;
  cursor: pointer;
}

.close-button {
  float: left;
  margin: 5px;
  padding: 2% 6%;
}

.login-title {
  font-size: 22px;
}

.button-container {
  position: relative;
  text-align: center;
  margin: 12% auto;
}

.loginprompt-button,
.signupprompt-button {
  margin: 20px;
  padding: 3% 8%;
  display: inline-block;
  font-size: 20px;
  font-weight: bold;
}

.loginprompt-button:hover,
.signupprompt-button:hover {
  background-color: $salmon;
  color: white;
}

.close-button:hover {
  background-color: $text-color;
  color: $off-white;
}

@media screen and (max-width: 1080px) {
  .LoginPrompt {
    left: 30%;
  }
}

@media screen and (max-width: 720px) {
  .LoginPrompt {
    left: 19%;
  }
}

@media screen and (max-width: 550px) {
  .LoginPrompt {
    left: 11%;
    width: 80%;
    height: 160px;
  }

  .close-button {
    float: left;
    margin: 5px;
    padding: 1% 2%;
  }
  
  .login-title {
    font-size: 16px;
    padding-top: 11px;
  }

  .loginprompt-button,
  .signupprompt-button {
    margin: 10px 15px;
    padding: 2% 8%;
  }
}