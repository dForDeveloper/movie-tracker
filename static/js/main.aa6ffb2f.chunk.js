(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{30:function(e,t,a){e.exports=a(45)},35:function(e,t,a){},45:function(e,t,a){"use strict";a.r(t);var n=a(0),r=a.n(n),s=a(24),i=a.n(s),o=(a(35),a(7)),c=a(8),l=a(10),u=a(9),m=a(11),p=a(6),d=a(46),v=a(47),h=a(50),f=function(e){return{type:"SET_USER",user:e}},g=function(e){return{type:"SET_FAVORITES",favorites:e}},b=function(e){return{type:"TOGGLE_LOGIN_PROMPT",validity:e}},E=a(2),w=a.n(E),O=a(4),j=function(){var e=Object(O.a)(w.a.mark(function e(t){var a,n,r=arguments;return w.a.wrap(function(e){for(;;)switch(e.prev=e.next){case 0:return a=r.length>1&&void 0!==r[1]?r[1]:{},e.next=3,fetch(t,a);case 3:if(!(n=e.sent).ok){e.next=10;break}return e.next=7,n.json();case 7:return e.abrupt("return",e.sent);case 10:throw Error("".concat(n.statusText," - Status ").concat(n.status));case 11:case"end":return e.stop()}},e,this)}));return function(t){return e.apply(this,arguments)}}(),y=a(16),k=a(49),N=function(e){return function(){var t=Object(O.a)(w.a.mark(function t(a){var n,r,s;return w.a.wrap(function(t){for(;;)switch(t.prev=t.next){case 0:return n="https://movie-tracker-jd.herokuapp.com/api/users/".concat(e,"/favorites"),t.next=3,j(n);case 3:r=t.sent,s=r.data.map(function(e){return e.movie_id}),a(g(s)),localStorage.setItem("favorites",JSON.stringify(s));case 7:case"end":return t.stop()}},t,this)}));return function(e){return t.apply(this,arguments)}}()},S=function(e,t){return function(){var a=Object(O.a)(w.a.mark(function a(n){var r,s,i,o,c;return w.a.wrap(function(a){for(;;)switch(a.prev=a.next){case 0:return a.prev=0,"https://movie-tracker-jd.herokuapp.com/api/users",r={method:"POST",body:JSON.stringify({email:e,password:t}),headers:{"Content-Type":"application/json"}},a.next=5,j("https://movie-tracker-jd.herokuapp.com/api/users",r);case 5:return s=a.sent,i=s.data,o=i.name,c=i.id,n(f({name:o,id:c})),a.next=10,n(N(c));case 10:return n(b(!1)),localStorage.setItem("user",JSON.stringify({name:o,id:c})),a.abrupt("return",s.status);case 15:return a.prev=15,a.t0=a.catch(0),a.abrupt("return","error");case 18:case"end":return a.stop()}},a,this,[[0,15]])}));return function(e){return a.apply(this,arguments)}}()},C=function(e){function t(){var e;return Object(o.a)(this,t),(e=Object(l.a)(this,Object(u.a)(t).call(this))).handleChange=function(t){var a=t.target,n=a.id,r=a.value;e.setState(Object(y.a)({},n,r))},e.handleSubmit=function(){var t=Object(O.a)(w.a.mark(function t(a){var n,r,s,i;return w.a.wrap(function(t){for(;;)switch(t.prev=t.next){case 0:return a.preventDefault(),n=e.state,r=n.email,s=n.password,t.next=4,e.props.loginUser(r,s);case 4:i=t.sent,e.setState({status:i});case 6:case"end":return t.stop()}},t,this)}));return function(e){return t.apply(this,arguments)}}(),e.state={email:"",password:"",status:""},e}return Object(m.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){var e=this.state.status;return r.a.createElement("form",{onSubmit:this.handleSubmit,className:"LoginForm"},r.a.createElement("h2",{className:"login-form-title"},"Log In"),r.a.createElement("label",{htmlFor:"email"},"Email:"),r.a.createElement("input",{type:"email",id:"email",className:"email-input",onChange:this.handleChange}),r.a.createElement("label",{htmlFor:"password"},"Password:"),r.a.createElement("input",{type:"password",id:"password",onChange:this.handleChange}),r.a.createElement("input",{className:"login-button",type:"submit",value:"login"}),"error"===e&&r.a.createElement("p",{className:"notice"},"Email and password do not match."),"success"===e&&r.a.createElement(k.a,{to:"/"}))}}]),t}(n.Component),x=Object(p.b)(null,function(e){return{loginUser:function(t,a){return e(S(t,a))}}})(C),P=Object(p.b)(null,function(e){return{toggleLoginPrompt:function(t){return e(b(t))}}})(function(e){var t=e.toggleLoginPrompt;return r.a.createElement("div",{className:"LoginPrompt"},r.a.createElement("button",{className:"close-button",onClick:function(){return t(!1)}},"X"),r.a.createElement("div",{className:"button-container"},r.a.createElement("h3",{className:"login-title"},"You must be logged in to save favorites."),r.a.createElement(d.a,{to:"/sign-up"},r.a.createElement("button",{className:"signupprompt-button",onClick:function(){return t(!1)}},"Sign Up")),r.a.createElement(d.a,{to:"/login"},r.a.createElement("button",{className:"loginprompt-button",onClick:function(){return t(!1)}},"Log In"))))}),T=a(26),L=function(e){function t(){var e,a;Object(o.a)(this,t);for(var n=arguments.length,r=new Array(n),s=0;s<n;s++)r[s]=arguments[s];return(a=Object(l.a)(this,(e=Object(u.a)(t)).call.apply(e,[this].concat(r)))).handleClick=function(){var e=Object(O.a)(w.a.mark(function e(t){var n;return w.a.wrap(function(e){for(;;)switch(e.prev=e.next){case 0:if(!t.name){e.next=6;break}return n=Object(T.a)({},a.props,{user_id:t.id,movie_id:a.props.id}),e.next=4,a.props.toggleFavorite(n,a.props.favorite);case 4:e.next=7;break;case 6:a.props.toggleLoginPrompt(!0);case 7:case"end":return e.stop()}},e,this)}));return function(t){return e.apply(this,arguments)}}(),a}return Object(m.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){var e=this,t=this.props,a=t.title,n=t.poster_path,s=t.currentUser,i=t.favorite,o=t.id;return r.a.createElement("div",{className:"MovieCard"},r.a.createElement("h3",{className:"moviecard-title"},a),r.a.createElement(d.a,{to:"/movies/".concat(o)},r.a.createElement("img",{src:"https://image.tmdb.org/t/p/w500".concat(n),alt:a,className:"moviecard-image"})),r.a.createElement("div",{className:"moviecard-favorite",onClick:function(){return e.handleClick(s)}},i?"Remove \u274c":"Favorite \ud83d\udc99"))}}]),t}(n.Component),F=Object(p.b)(function(e){return{currentUser:e.currentUser}},function(e){return{toggleLoginPrompt:function(t){return e(b(t))},toggleFavorite:function(t,a){return e(function(e,t){return function(){var a=Object(O.a)(w.a.mark(function a(n){var r,s,i,o,c;return w.a.wrap(function(a){for(;;)switch(a.prev=a.next){case 0:if(r=e.user_id,s=e.movie_id,!t){a.next=8;break}return i="https://movie-tracker-jd.herokuapp.com/api/users/".concat(r,"/favorites/").concat(s),o={method:"DELETE",body:JSON.stringify({user_id:r,movie_id:s}),headers:{"Content-Type":"application/json"}},a.next=6,j(i,o);case 6:a.next=12;break;case 8:return c={method:"POST",body:JSON.stringify(e),headers:{"Content-Type":"application/json"}},a.next=12,j("https://movie-tracker-jd.herokuapp.com/api/users/favorites/new",c);case 12:n(N(r));case 13:case"end":return a.stop()}},a,this)}));return function(e){return a.apply(this,arguments)}}()}(t,a))}}})(L),U=function(e){function t(){var e,a;Object(o.a)(this,t);for(var n=arguments.length,s=new Array(n),i=0;i<n;i++)s[i]=arguments[i];return(a=Object(l.a)(this,(e=Object(u.a)(t)).call.apply(e,[this].concat(s)))).getMoviesToDisplay=function(e){var t=a.props.favorites;return e.map(function(e){return t.includes(e.id)?r.a.createElement(F,Object.assign({key:e.id},e,{favorite:!0})):r.a.createElement(F,Object.assign({key:e.id},e,{favorite:!1}))})},a}return Object(m.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){var e=this.props,t=e.movies,a=e.favorites,n=e.match,s=e.showLoginPrompt,i=t.filter(function(e){return a.includes(e.id)});return r.a.createElement("div",{className:"MovieContainer"},s&&r.a.createElement(P,null),"/"===n.path&&this.getMoviesToDisplay(t),"/favorites"===n.path&&this.getMoviesToDisplay(i),"/favorites"===n.path&&0===a.length&&r.a.createElement("p",null,"There are no favorites to display."))}}]),t}(n.Component),_=Object(p.b)(function(e){return{movies:e.movies,favorites:e.favorites,showLoginPrompt:e.showLoginPrompt}})(U),I=function(e){var t=e.title,a=e.release_date,n=e.poster_path,s=e.vote_average,i=e.overview;return r.a.createElement("div",{className:"MovieDetails"},r.a.createElement("img",{src:"https://image.tmdb.org/t/p/w500".concat(n),alt:t,className:"movie-image"}),r.a.createElement("div",{className:"movie-info"},r.a.createElement("h3",{className:"movie-title"},t),r.a.createElement("p",{className:"movie-overview"}," ",i),r.a.createElement("p",{className:"movie-details"},r.a.createElement("span",{className:"detail-bold"},"Release Date: "),a),r.a.createElement("p",{className:"movie-details"},r.a.createElement("span",{className:"detail-bold"},"Rating: "),s)))},M=function(e){function t(){var e,a;Object(o.a)(this,t);for(var n=arguments.length,s=new Array(n),i=0;i<n;i++)s[i]=arguments[i];return(a=Object(l.a)(this,(e=Object(u.a)(t)).call.apply(e,[this].concat(s)))).handleClick=function(){a.props.setUser({}),a.props.setFavorites([]),localStorage.removeItem("user"),localStorage.removeItem("favorites")},a.getUserLinks=function(){var e=a.props,t=e.currentUser,n=e.location;return t.name?r.a.createElement("div",{className:"user-links"},r.a.createElement("p",{className:"greeting"},"Hello,",r.a.createElement("span",{className:"username"},t.name)),r.a.createElement("span",{className:"nav-left"},"/"!==n.pathname&&r.a.createElement(d.a,{to:"/",id:"item-left"},"Home"),"/"===n.pathname&&r.a.createElement(d.a,{to:"/favorites",id:"item-left"},"View Favorites")),r.a.createElement("span",{className:"nav-right"},r.a.createElement(d.a,{to:"/",id:"logout-link",onClick:a.handleClick},"Log Out"))):r.a.createElement("div",{className:"user-links"},r.a.createElement("span",{className:"nav-left"},"/"!==n.pathname&&r.a.createElement(d.a,{to:"/",id:"item-left"},"Home"),r.a.createElement(d.a,{to:"/sign-up",id:"item-left"},"Sign Up"),r.a.createElement(d.a,{to:"/login",id:"item-right"},"Log In")),r.a.createElement("span",{className:"nav-right"},r.a.createElement("p",null,"Log in to save your favorite movies.")))},a}return Object(m.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){return r.a.createElement("nav",{className:"NavBar"},this.getUserLinks())}}]),t}(n.Component),D=Object(p.b)(function(e){return{currentUser:e.currentUser}},function(e){return{setUser:function(t){return e(f(t))},setFavorites:function(t){return e(g(t))}}})(M),R=function(e){function t(){var e;return Object(o.a)(this,t),(e=Object(l.a)(this,Object(u.a)(t).call(this))).handleChange=function(t){var a=t.target,n=a.id,r=a.value;e.setState(Object(y.a)({},n,r))},e.handleSubmit=function(){var t=Object(O.a)(w.a.mark(function t(a){var n,r,s,i,o;return w.a.wrap(function(t){for(;;)switch(t.prev=t.next){case 0:if(a.preventDefault(),n=e.state,r=n.name,s=n.email,i=n.passwordOriginal,!e.checkMatchingPassword()||!e.checkEmailRegex(s)){t.next=7;break}return t.next=5,e.props.createUser(r,s,i);case 5:o=t.sent,e.setState({status:o});case 7:case"end":return t.stop()}},t,this)}));return function(e){return t.apply(this,arguments)}}(),e.checkEmailRegex=function(e){return/[a-z]\w+@[a-z]\w+\.[a-z]\w+/.test(e)},e.checkMatchingPassword=function(){var t=e.state;return t.passwordOriginal===t.passwordConfirmed},e.getNameAndEmailInputFields=function(){var t=e.state,a=t.name,n=t.email,s=e.checkEmailRegex(n)?null:"input--red";return r.a.createElement("div",{className:"div--email"},r.a.createElement("label",{htmlFor:"name"},"Name:"),r.a.createElement("input",{id:"name",value:a,onChange:e.handleChange,required:!0}),r.a.createElement("label",{htmlFor:"email"},"Email"),r.a.createElement("input",{id:"email",value:n,onChange:e.handleChange,required:!0,className:s}),!e.checkEmailRegex(n)&&r.a.createElement("p",{className:"notice"},"Please enter a valid email."))},e.getPasswordInputFields=function(){var t=e.state,a=t.passwordOriginal,n=t.passwordConfirmed,s=e.checkMatchingPassword()?null:"input--red";return r.a.createElement("div",{className:"div--password"},r.a.createElement("label",{htmlFor:"passwordOriginal"},"Password"),r.a.createElement("input",{id:"passwordOriginal",value:a,onChange:e.handleChange,type:"password",required:!0}),r.a.createElement("label",{htmlFor:"passwordConfirmed"},"Confirm Password"),r.a.createElement("input",{id:"passwordConfirmed",value:n,onChange:e.handleChange,type:"password",required:!0,className:s}),!e.checkMatchingPassword()&&r.a.createElement("p",{className:"notice"},"Passwords do not match."))},e.state={name:"",email:"",passwordOriginal:"",passwordConfirmed:"",status:""},e}return Object(m.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){var e=this.state.status;return r.a.createElement("form",{onSubmit:this.handleSubmit,className:"SignUpForm"},r.a.createElement("h2",{className:"signup-title"},"Sign Up"),this.getNameAndEmailInputFields(),this.getPasswordInputFields(),r.a.createElement("input",{className:"create-button",type:"submit",value:"Submit"}),"error"===e&&r.a.createElement("p",{className:"notice"},"Email has already been used."),"success"===e&&r.a.createElement(k.a,{to:"/"}))}}]),t}(n.Component),A=Object(p.b)(null,function(e){return{createUser:function(t,a,n){return e(function(e,t,a){return function(){var n=Object(O.a)(w.a.mark(function n(r){var s,i;return w.a.wrap(function(n){for(;;)switch(n.prev=n.next){case 0:return n.prev=0,s={method:"POST",body:JSON.stringify({name:e,email:t,password:a}),headers:{"Content-Type":"application/json"}},n.next=5,j("https://movie-tracker-jd.herokuapp.com/api/users/new",s);case 5:return i=n.sent,n.next=8,r(S(t,a));case 8:return n.abrupt("return",i.status);case 11:return n.prev=11,n.t0=n.catch(0),n.abrupt("return","error");case 14:case"end":return n.stop()}},n,this,[[0,11]])}));return function(e){return n.apply(this,arguments)}}()}(t,a,n))}}})(R),J=function(e){function t(){var e,a;Object(o.a)(this,t);for(var n=arguments.length,r=new Array(n),s=0;s<n;s++)r[s]=arguments[s];return(a=Object(l.a)(this,(e=Object(u.a)(t)).call.apply(e,[this].concat(r)))).componentDidMount=function(){var e="https://api.themoviedb.org/3/movie/upcoming?api_key=".concat("d34f1928ae0d9072cc089d91046e5c3d");a.props.fetchMovies(e),a.getLocalStorage()},a.getLocalStorage=function(){var e=JSON.parse(localStorage.getItem("user")),t=JSON.parse(localStorage.getItem("favorites"));e&&a.props.setUser(e),t&&a.props.setFavorites(t)},a}return Object(m.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){var e=this.props.movies;return r.a.createElement("div",{className:"App"},r.a.createElement("header",{className:"title-section"},r.a.createElement(d.a,{to:"/"},r.a.createElement("h1",{className:"app-title"},"Movie Tracker")),r.a.createElement("h2",{className:"app-subtitle"},"Find and save your favorite new releases.")),r.a.createElement(v.a,{path:"/",component:D}),r.a.createElement(v.a,{path:"/sign-up",component:A}),r.a.createElement(v.a,{path:"/login",component:x}),r.a.createElement(v.a,{path:"/favorites",component:_}),r.a.createElement(v.a,{exact:!0,path:"/",component:_}),r.a.createElement(v.a,{path:"/movies/:id",render:function(t){var a=t.match.params.id,n=e.find(function(e){return e.id===parseInt(a)});return n?r.a.createElement(I,n):null}}))}}]),t}(n.Component),G=Object(h.a)(Object(p.b)(function(e){return{movies:e.movies}},function(e){return{fetchMovies:function(t){return e(function(e){return function(){var t=Object(O.a)(w.a.mark(function t(a){var n;return w.a.wrap(function(t){for(;;)switch(t.prev=t.next){case 0:return t.next=2,j(e);case 2:n=t.sent,a({type:"ADD_MOVIES",movies:n.results});case 4:case"end":return t.stop()}},t,this)}));return function(e){return t.apply(this,arguments)}}()}(t))},setUser:function(t){return e(f(t))},setFavorites:function(t){return e(g(t))}}})(J));Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));var V=a(13),q=Object(V.combineReducers)({movies:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:[],t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"ADD_MOVIES":return t.movies;default:return e}},currentUser:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:{},t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"SET_USER":return t.user;default:return e}},favorites:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:[],t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"SET_FAVORITES":return t.favorites;default:return e}},showLoginPrompt:function(){var e=arguments.length>0&&void 0!==arguments[0]&&arguments[0],t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"TOGGLE_LOGIN_PROMPT":return t.validity;default:return e}}}),z=a(48),B=a(28),H=a(29),W=Object(B.composeWithDevTools)(Object(V.applyMiddleware)(H.a)),X=Object(V.createStore)(q,W);i.a.render(r.a.createElement(p.a,{store:X},r.a.createElement(z.a,null,r.a.createElement(G,null))),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then(function(e){e.unregister()})}},[[30,2,1]]]);
//# sourceMappingURL=main.aa6ffb2f.chunk.js.map