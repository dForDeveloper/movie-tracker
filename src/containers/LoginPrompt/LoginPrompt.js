import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { toggleLoginPrompt } from '../../actions';

export const LoginPrompt = (props) => {
  const { toggleLoginPrompt } = props;
  return (
    <div className="LoginPrompt">
      <button className="close-button" onClick={() => toggleLoginPrompt(false)}>X</button>
      <div className="button-container">
        <h3 className="login-title">
          You must be logged in to save favorites.
        </h3>
        <Link to='/sign-up'>
          <button className="signupprompt-button" onClick={() => toggleLoginPrompt(false)}>Sign Up</button>
        </Link>
        <Link to='/login'>
          <button className="loginprompt-button" onClick={() => toggleLoginPrompt(false)}>Log In</button>
        </Link>
      </div>
    </div>
  );
}

export const mapDispatchToProps = (dispatch) => ({
  toggleLoginPrompt: (validity) => dispatch(toggleLoginPrompt(validity))
})

export default connect(null, mapDispatchToProps)(LoginPrompt);

LoginPrompt.propTypes = {
  toggleLoginPrompt: PropTypes.func
}