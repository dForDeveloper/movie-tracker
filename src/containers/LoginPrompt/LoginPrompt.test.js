import React from 'react';
import { shallow } from 'enzyme';
import { LoginPrompt, mapDispatchToProps } from './LoginPrompt.js';
import { toggleLoginPrompt } from '../../actions';

describe('LoginPrompt', () => {
  let wrapper;
  const toggleLoginPromptMock = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <LoginPrompt toggleLoginPrompt={toggleLoginPromptMock} />
    );
  });

  describe('LoginPrompt component', () => {
    it('should properly render the component elements', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should call toggleLoginPrompt with false when the close button is clicked', () => {
      wrapper.find('.close-button').simulate('click');
      expect(toggleLoginPromptMock).toHaveBeenCalledWith(false);
    });
    
    it('should call toggleLoginPrompt with false when the sign up button is clicked', () => {
      wrapper.find('.signupprompt-button').simulate('click');
      expect(toggleLoginPromptMock).toHaveBeenCalledWith(false);
    });
    
    it('should call toggleLoginPrompt with false when the login button is clicked', () => {
      wrapper.find('.loginprompt-button').simulate('click');
      expect(toggleLoginPromptMock).toHaveBeenCalledWith(false);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the action toggleLoginPrompt', () => {
      const expectedAction = toggleLoginPrompt(true);
      const dispatchMock = jest.fn();
      const props = mapDispatchToProps(dispatchMock);
      props.toggleLoginPrompt(true);
      expect(dispatchMock).toHaveBeenCalledWith(expectedAction);
    });
  });
});
