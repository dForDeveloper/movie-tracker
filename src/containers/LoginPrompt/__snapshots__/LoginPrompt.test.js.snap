// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPrompt LoginPrompt component should properly render the component elements 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
